mlir/lib/Catalyst/Transforms/catalyst_to_llvm.cpp

```diff
@@ -435,14 +435,34 @@ struct PythonCallOpPattern : public OpConversionPattern<PythonCallOp> {
         rewriter.setInsertionPointToStart(mod.getBody());
 
         Type i64 = rewriter.getI64Type();
-        LLVM::LLVMFuncOp customCallFnOp =
-            mlir::LLVM::lookupOrCreateFn(mod, "pyregistry", {/*args=*/i64}, /*ret_type=*/voidType);
+        // The argument convention is as follows:
+        // arg0 = identifier
+        // arg1 = length of varargs
+        // arg2..N+2 varargs pointers to memrefs
+        bool isVarArg = true;
+        LLVM::LLVMFuncOp customCallFnOp = mlir::LLVM::lookupOrCreateFn(
+            mod, "pyregistry", {/*args=*/i64, i64}, /*ret_type=*/voidType, isVarArg);
```

Comment:
So after the two integers, it can be any type arguments because it is variadic?

Reply:
It will be guaranteed at compile time that var args are pointers to memrefs. But that cannot be expressed here. So anything after the second parameter is a pointer to a memref.